spec/index.html

```diff
@@ -1493,3 +1503,221 @@ <h1>SharedArray ( ..._values_ )</h1>
   </emu-clause>
   </ins>
 </emu-clause>
+
+<emu-clause id="sec-synchronization-primitives">
+  <h1>Synchronization Primitives</h1>
+  <p>Mutexes and condition variables are provided as higher level abstractions, as an easier to use alternative to user-built abstractions on top of Atomics.wait and Atomics.notify. They are Shared Structs with no fields.</p>
+
+  <ins class="block">
+  <emu-clause id="sec-abstract-operations-for-mutex">
+    <h1>Abstract Operations for Mutex Objects</h1>
+
+    <emu-clause id="sec-unlocktokencreateifneeded" type="abstract operation">
+      <h1>
+        UnlockTokenCreateIfNeeded (
+          _token_: an Object or *undefined*,
+          _mutex_: an Object,
+        ) : an Object
+      </h1>
+      <dl class="header">
+      </dl>
+      <emu-alg>
+        1. Assert: _mutex_ has a [[MutexWaiterList]] internal slot.
+        1. If _token_ is *undefined*, then
+          1. Set _token_ to OrdinaryObjectCreate(%MutexUnlockTokenPrototype%, « [[LockedMutex]] »).
+        1. Assert: _token_ has a [[LockedMutex]] internal slot.
```

Comment:
Probably also assert that a passed token is not currently locked?
```suggestion
        1. If _token_ is *undefined*, then
          1. Set _token_ to OrdinaryObjectCreate(%MutexUnlockTokenPrototype%, « [[LockedMutex]] »).
```suggestion
        1. If _token_ is *undefined*, then
          1. Set _token_ to OrdinaryObjectCreate(%MutexUnlockTokenPrototype%, « [[LockedMutex]] »).
        1. Else,
          1. Assert: _token_ has a [[LockedMutex]] internal slot.
          1. Assert: _token_.[[LockedMutex]] is ~empty~.
``````